needle/modules/configuration/mdm_assess.py

```diff
@@ -0,0 +1,130 @@
+from core.framework.module import BaseModule
+from core.framework.framework import FrameworkException
+from core.device.device import Device
+from core.utils.utils import Utils
+from core.utils.constants import Constants
+import re, sys, plistlib
+
+class Module(BaseModule):
+    meta = {
+        'name': 'MDM Assess',
+        'author': 'Oliver Simonnet (@MWRLabs)',
+        'description': 'Automated MDM Configuration Assessment Module.',
+        'options': (
+            ('template', True, True, 'Configuration template.[Plist]'),
+            ('mode', False, True, 'Output mode [1|2|3].')
+        ),
+        'comments': [
+            'MODE: 1 Displays misconfigurations only.',
+            'MODE: 2 Displays recomendations.',
+            'MODE: 3 Displats all configutaion.',
+            'TEMPLATE: This is an EffectiveUserSettings.plist file containing you desired configuration.']
+    }
+
+    # ==================================================================================================================
+    # UTILS
+    # ==================================================================================================================
+    def __init__(self, params):
+        BaseModule.__init__(self, params)
+        # Setting default output file
+        self.options['mode'] = 1
+
+    # Constructs ouput filename
+    def set_output_name(self, remote_file):
+        file_name = Utils.extract_filename_from_path(remote_file)
+        file_name = 'mdm_assess_{}'.format(file_name)
+        return self.local_op.build_output_path_for_file(file_name, self)
+
+    # Check if file is a valid Plist
+    def is_plist(self, config_file):
+        try: 
+            plistlib.readPlist(config_file)
+            return True
+        except: 
+            # If incorrect format, print error and exit
+            self.printer.error("Invalid configuration file!")
+            self.printer.verbose("Invalid file: %s" % config_file)
+            raise FrameworkException()
+        
+    # Parse Plist configuration data into dict
+    def parse_config_data(self, config_file):
+        if self.is_plist(config_file):
+            config, merged = plistlib.readPlist(config_file), {}
```

Comment:
please use `Utils.plist_read_from_file` or `Utils.plist_read_from_string` to read a plist file